### ISSUE TITLE
saliency is NaN for VGG16 like model with BatchNorm

### ISSUE THREAD

**Issue by sverneka** [jacobgil/keras-grad-cam#19]
saliency = saliency_fn([preprocessed_input, 0]) this gives me NaN for valid inputs. Below is my modified VGG16 network with BatchNorm.

def VGG16(input_shape=(224,224,3),classes=10):
    input_img = Input(shape=input_shape)
    # Block 1
    x=Conv2D(64, (3, 3), activation='relu', padding='same', input_shape= input_shape, name='block1_conv1', kernel_initializer='glorot_normal')(input_img)
    x=Conv2D(64, (3, 3), activation='relu', padding='same', name='block1_conv2', kernel_initializer='glorot_normal')(x)
    x=MaxPooling2D((2, 2), strides=(2, 2), name='block1_pool')(x)
    x=BatchNormalization()(x)
    # Block 2
    x=Conv2D(128, (3, 3), activation='relu', padding='same', name='block2_conv1', kernel_initializer='glorot_normal')(x)
    x=Conv2D(128, (3, 3), activation='relu', padding='same', name='block2_conv2', kernel_initializer='glorot_normal')(x)
    x=MaxPooling2D((2, 2), strides=(2, 2), name='block2_pool')(x)
    x=BatchNormalization()(x)
    # Block 3
    x=Conv2D(256, (3, 3), activation='relu', padding='same', name='block3_conv1', kernel_initializer='glorot_normal')(x)
    x=Conv2D(256, (3, 3), activation='relu', padding='same', name='block3_conv2', kernel_initializer='glorot_normal')(x)
    x=Conv2D(256, (3, 3), activation='relu', padding='same', name='block3_conv3', kernel_initializer='glorot_normal')(x)
    x=MaxPooling2D((2, 2), strides=(2, 2), name='block3_pool')(x)
    x=BatchNormalization()(x)
    # Block 4
    x=Conv2D(512, (3, 3), activation='relu', padding='same', name='block4_conv1', kernel_initializer='glorot_normal')(x)
    x=Conv2D(512, (3, 3), activation='relu', padding='same', name='block4_conv2', kernel_initializer='glorot_normal')(x)
    x=Conv2D(512, (3, 3), activation='relu', padding='same', name='block4_conv3', kernel_initializer='glorot_normal')(x)
    x=MaxPooling2D((2, 2), strides=(2, 2), name='block4_pool')(x)
    x=BatchNormalization()(x)
    # Block 5
    x=Conv2D(512, (3, 3), activation='relu', padding='same', name='block5_conv1', kernel_initializer='glorot_normal')(x)
    x=Conv2D(512, (3, 3), activation='relu', padding='same', name='block5_conv2', kernel_initializer='glorot_normal')(x)
    x=Conv2D(512, (3, 3), activation='relu', padding='same', name='block5_conv3', kernel_initializer='glorot_normal')(x)
    x=MaxPooling2D((2, 2), strides=(2, 2), name='block5_pool')(x)
    # Classification block
    x=Flatten(name='flatten')(x)
    x=Dense(256, activation='relu', name='fc1', kernel_initializer='glorot_normal')(x)
    #model.add(Dropout(0.5))
    x=Dense(128, activation='relu', name='fc2', kernel_initializer='glorot_normal')(x)
    #model.add(Dropout(0.5))
    x=Dense(classes, activation='softmax', name='predictions', kernel_initializer='glorot_normal')(x)
    model = Model(input_img, x)
    model.summary()
    #model = Model(inputs, x, name='vgg16')
    return model

**Comment by sverneka** [comment]
There was an issue with the model itself, probably some weights were NaN. The issue is now resolved.
